### PR TITLE
 bpo-35960: Fix dataclasses.field throwing away empty metadata. 

### PR DESCRIPTION
**Pull request by chrahunt**
https://bugs.python.org/issue35960

### CODE REVIEW

File: Lib/test/test_dataclasses.py [hunk 1]
```diff
@@ -1737,23 +1737,25 @@ class C:
                 i: int = field(metadata=0)
 
         # Make sure an empty dict works.
+        d = {}
```
Comment (remilapeyre) [inline Lib/test/test_dataclasses.py:1740]:
```suggestion
        # See issue 35960
        d = {} 
```

**Review by remilapeyre — APPROVED** [review]

File: Lib/test/test_dataclasses.py [hunk 2]
```diff
@@ -1737,23 +1737,25 @@ class C:
...
         with self.assertRaisesRegex(TypeError,
                                     'does not support item assignment'):
             fields(C)[0].metadata['test'] = 3
 
         # Make sure a non-empty dict works.
+        d = {'test': 10, 'bar': '42', 3: 'three'}
         @dataclass
         class C:
-            i: int = field(metadata={'test': 10, 'bar': '42', 3: 'three'})
-        self.assertEqual(len(fields(C)[0].metadata), 3)
```
Comment (ericvsmith) [inline Lib/test/test_dataclasses.py]: My intent with putting explicit checks for the length and values here was to ensure that if the original mapping was changed by MappingProxy, these test would detect that. I realize it's a low probability test, but I'd like to leave it in.

Comment (eamanu) [reply Lib/test/test_dataclasses.py]: Yes, I'd like to leave it in too.

File: Lib/test/test_dataclasses.py (hunk 1 above)
Comment (chrahunt) [reply Lib/test/test_dataclasses.py:1740]: Thanks, I have addressed this in e302e24.

File: Lib/test/test_dataclasses.py (hunk 2 above)
Comment (chrahunt) [reply Lib/test/test_dataclasses.py]: That makes sense, I have added the tests back in e302e24.

**Pull request merged**